server/controllers/stock/index.js

```diff
@@ -298,96 +298,91 @@ async function createInventoryAdjustment(req, res, next) {
+      if (difference < 0) {
+        // we have to do a stock exit movement
+        // we must substract the |difference| to the current quantity
+        countNeedDecrease++;
+        const movementObject = {
+          uuid : db.bid(uuid()),
+          lot_uuid : db.bid(lot.uuid),
+          depot_uuid : db.bid(movement.depot_uuid),
+          document_uuid : db.bid(uniqueAdjustmentUuid),
+          quantity : Math.abs(difference),
+          unit_cost : lot.unit_cost,
+          date : new Date(movement.date),
+          entity_uuid : movement.entity_uuid,
+          is_exit : 1,
+          flux_id : core.flux.INVENTORY_ADJUSTMENT,
+          description : movement.description,
+          user_id : req.session.user.id,
+          period_id : periodId,
+        };
+        adjustmentLog.push(_.assign({}, lot, { difference, is_exit : 1 }));
+        trx.addQuery('INSERT INTO stock_movement SET ?', movementObject);
+      } else {
+        // we must do increase the current quantity by the |difference|
+        countNeedIncrease++;
+        const movementObject = {
+          uuid : db.bid(uuid()),
+          lot_uuid : db.bid(lot.uuid),
+          depot_uuid : db.bid(movement.depot_uuid),
+          document_uuid : db.bid(uniqueAdjustmentUuid),
+          quantity : Math.abs(difference),
+          unit_cost : lot.unit_cost,
+          date : new Date(movement.date),
+          entity_uuid : movement.entity_uuid,
+          is_exit : 0,
+          flux_id : core.flux.INVENTORY_ADJUSTMENT,
+          description : movement.description,
+          user_id : req.session.user.id,
+          period_id : periodId,
+        };
+        adjustmentLog.push(_.assign({}, lot, { difference, is_exit : 0 }));
+        trx.addQuery('INSERT INTO stock_movement SET ?', movementObject);
+      }
     });
 
-    const negativeAdjustmentParams = [
-      db.bid(negativeAdjustmentUuid), 1, req.session.project.id,
+    const decreaseParams = [
+      db.bid(uniqueAdjustmentUuid), 1, req.session.project.id,
     ];
 
-    const positiveAdjustmentParams = [
-      db.bid(positiveAdjustmentUuid), 0, req.session.project.id,
+    const increaseParams = [
+      db.bid(uniqueAdjustmentUuid), 0, req.session.project.id,
     ];
 
     if (req.session.stock_settings.enable_auto_stock_accounting) {
-      if (positiveQuantities.length > 0) {
-        trx.addQuery('CALL PostStockMovement(?)', [negativeAdjustmentParams]);
+      if (countNeedDecrease > 0) {
+        trx.addQuery('CALL PostStockMovement(?)', [decreaseParams]);
       }
 
-      if (negativeQuantities.length > 0) {
-        trx.addQuery('CALL PostStockMovement(?)', [positiveAdjustmentParams]);
+      if (countNeedIncrease > 0) {
+        trx.addQuery('CALL PostStockMovement(?)', [increaseParams]);
       }
     }
 
-    // reset all previous lots
-    await trx.execute();
+    const logParams = [
+      db.bid(uniqueAdjustmentUuid),
+      JSON.stringify(adjustmentLog),
+      req.session.user.id,
```

Comment:
Do you need the user on both this table and the `stock_movement` table?  It seems like you probably only need the `document_uuid` and you could recover the user, right?